Source/Meadow.Foundation.Peripherals/Sensors.Temperature.Max31865/Driver/Max31865.cs

```diff
@@ -0,0 +1,425 @@
+    /// <param name="filterMode">Noise rejection filter mode.</param>
+    public Max31865(ISpiBus bus, IPin? chipSelect, KnownSensorType knownSensorType, Wires wires, int referenceResistor, ConversionFilterMode filterMode)
+        : this(bus, chipSelect?.CreateDigitalOutputPort(true), knownSensorType, wires, referenceResistor, filterMode)
+    {
+        portCreated = true;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Max31865"/> class using a digital output port for chip select.
+    /// </summary>
+    /// <param name="bus">The SPI bus instance.</param>
+    /// <param name="chipSelect">The chip select digital output port. If null, chip select must be managed externally.</param>
+    /// <param name="knownSensorType">The type of Platinum Resistance Thermometer.</param>
+    /// <param name="wires">The number of wires the Platinum Resistance Thermometer has.</param>
+    /// <param name="referenceResistor">The reference resistor value in Ohms.</param>
+    /// <param name="filterMode">Noise rejection filter mode.</param>
+    public Max31865(ISpiBus bus, IDigitalOutputPort? chipSelect, KnownSensorType knownSensorType, Wires wires, int referenceResistor, ConversionFilterMode filterMode)
+    {
+        port = chipSelect;
+        spiComms = new SpiCommunications(bus, chipSelect, DefaultSpiBusSpeed);
+        this.knownSensorType = knownSensorType;
+        this.wires = wires;
+        rtdConverter = new DirectMathematicalMethod();
+        conversionFilterMode = filterMode;
+
+        Initialize();
+    }
+
+    /// <summary>
+    /// Gets or sets the SPI bus mode for communication.
+    /// </summary>
+    public SpiClockConfiguration.Mode SpiBusMode
+    {
+        get => spiComms.BusMode;
+        set => spiComms.BusMode = value;
+    }
+
+    /// <summary>
+    /// Gets or sets the SPI bus speed for communication.
+    /// </summary>
+    public Frequency SpiBusSpeed
+    {
+        get => spiComms.BusSpeed;
+        set => spiComms.BusSpeed = value;
+    }
+
+    /// <inheritdoc />
+    protected override async Task<Temperature> ReadSensor()
+    {
+        var rtd = await ReadInternal();
+
+        return new Temperature(this.rtdConverter.Convert(rtd, (float)knownSensorType, referenceResistor));
+    }
+
+    /// <summary>
+    /// Starts continuously sampling the sensor.
+    ///
+    /// This method also starts raising `Changed` events and IObservable
+    /// subscribers getting notified. Use the `readIntervalDuration` parameter
+    /// to specify how often events and notifications are raised/sent.
+    /// </summary>
+    /// <param name="updateInterval">A `TimeSpan` that specifies how long to
+    /// wait between readings. This value influences how often `*Updated`
+    /// events are raised and `IObservable` consumers are notified.
+    ///</param>
+    public override void StartUpdating(TimeSpan? updateInterval = null)
```

Comment:
You could also just derive from SamplingSensorBase<T>.

Reply:
I have moved away from sampling now